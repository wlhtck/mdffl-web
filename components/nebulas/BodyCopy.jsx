import ReactMarkdown from 'react-markdown';
import { oneOf, bool } from 'prop-types';
import { compose, setDisplayName } from 'recompose';
import withStyles from '../util/withStyles';
import styles from './fontStyles';

const BodyCopy = compose(
  setDisplayName('BodyCopy'),
  withStyles(({ inverted, align }) => styles({
    align, color: (inverted ? 'white' : 'black'), size: '24px',
  })),
)(ReactMarkdown);

BodyCopy.propTypes = {
  textAlign: oneOf(['left', 'right', 'center']),
  inverted: bool,
};

BodyCopy.defaultProps = {
  textAlign: 'left',
  inverted: false,
};

export default BodyCopy;
