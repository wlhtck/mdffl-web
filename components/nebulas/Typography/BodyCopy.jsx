import ReactMarkdown from 'react-markdown';
import { oneOf, bool } from 'prop-types';
import withStyles from '../withStyles';
import styles from './styles';

// TODO: Rename this parameter to align
const BodyCopy = withStyles(({ inverted, textAlign }) => styles({
  color: (inverted ? 'white' : 'black'), size: '24px', align: textAlign,
}), ReactMarkdown);

BodyCopy.propTypes = {
  textAlign: oneOf(['left', 'right', 'center']),
  inverted: bool,
};

BodyCopy.defaultProps = {
  textAlign: 'left',
  inverted: false,
};

export default BodyCopy;
