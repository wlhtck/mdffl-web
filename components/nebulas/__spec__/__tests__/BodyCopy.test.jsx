import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import BodyCopy from '../../Typography/BodyCopy';

describe('nebulas/Typography/BodyCopy', () => {
  it('renders a <BodyCopy />', () => {
    expect(mount(
      <BodyCopy>
        The Metro Detroit Flag Football League (MDFFL) is an LGBT and straight allied flag football
        league, changing preconceived societal beliefs about sexual orientation and
        self-identification.
      </BodyCopy>,
    )).toMatchSnapshot();
  });
});
