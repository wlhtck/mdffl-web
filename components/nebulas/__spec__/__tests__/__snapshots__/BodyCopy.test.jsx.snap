// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nebulas/Typography/BodyCopy renders a <BodyCopy /> 1`] = `
.c0 {
  color: #303133;
  font-family: 'Lato',sans-serif;
  font-weight: 600;
  font-size: 24px;
  text-align: left;
  margin: 0 0 32px;
}

<withStyles
  inverted={false}
  textAlign="left"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "withStyles-sc-8qn4cd-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            [Function],
          ],
        },
        "defaultProps": Object {
          "inverted": false,
          "textAlign": "left",
        },
        "displayName": "withStyles",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "inverted": [Function],
          "textAlign": [Function],
        },
        "render": [Function],
        "renderers": Object {
          "blockquote": "blockquote",
          "break": "br",
          "code": [Function],
          "definition": [Function],
          "delete": "del",
          "emphasis": "em",
          "heading": [Function],
          "html": [Function],
          "image": "img",
          "imageReference": "img",
          "inlineCode": [Function],
          "link": "a",
          "linkReference": "a",
          "list": [Function],
          "listItem": [Function],
          "paragraph": "p",
          "parsedHtml": [Function],
          "root": [Function],
          "strong": "strong",
          "table": [Function],
          "tableBody": [Function],
          "tableCell": [Function],
          "tableHead": [Function],
          "tableRow": [Function],
          "text": [Function],
          "thematicBreak": "hr",
          "virtualHtml": [Function],
        },
        "styledComponentId": "withStyles-sc-8qn4cd-0",
        "target": [Function],
        "toString": [Function],
        "types": Array [
          "break",
          "paragraph",
          "emphasis",
          "strong",
          "thematicBreak",
          "blockquote",
          "delete",
          "link",
          "image",
          "linkReference",
          "imageReference",
          "table",
          "tableHead",
          "tableBody",
          "tableRow",
          "tableCell",
          "root",
          "text",
          "list",
          "listItem",
          "definition",
          "heading",
          "inlineCode",
          "code",
          "html",
          "virtualHtml",
          "parsedHtml",
        ],
        "uriTransformer": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
    inverted={false}
    textAlign="left"
  >
    <ReactMarkdown
      astPlugins={Array []}
      className="c0"
      escapeHtml={true}
      inverted={false}
      plugins={Array []}
      rawSourcePos={false}
      renderers={Object {}}
      skipHtml={false}
      sourcePos={false}
      textAlign="left"
      transformLinkUri={[Function]}
    >
      <Root
        className="c0"
        key="root-1-1"
      >
        <div
          className="c0"
        >
          <p
            key="paragraph-1-1"
          >
            <TextRenderer
              key="text-1-1"
              nodeKey="text-1-1"
              value="The Metro Detroit Flag Football League (MDFFL) is an LGBT and straight allied flag football league, changing preconceived societal beliefs about sexual orientation and self-identification."
            >
              The Metro Detroit Flag Football League (MDFFL) is an LGBT and straight allied flag football league, changing preconceived societal beliefs about sexual orientation and self-identification.
            </TextRenderer>
          </p>
        </div>
      </Root>
    </ReactMarkdown>
  </StyledComponent>
</withStyles>
`;
